collections/src/main/scala/strawman/collection/immutable/WrappedString.scala

```diff
@@ -0,0 +1,59 @@
+package strawman.collection
+package immutable
+
+import scala.{Char, Int}
+import scala.Predef.String
+import mutable.{Builder, StringBuilder}
+
+/**
+  *  This class serves as a wrapper augmenting `String`s with all the operations
+  *  found in indexed sequences.
+  *
+  *  The difference between this class and `StringOps` is that calling transformer
+  *  methods such as `filter` and `map` will yield an object of type `WrappedString`
+  *  rather than a `String`.
+  *
+  *  @param self    a string contained within this wrapped string
+  *
+  *  @since 2.8
+  *  @define Coll `WrappedString`
+  *  @define coll wrapped string
+  */
+final class WrappedString(val self: String) extends AbstractSeq[Char] with IndexedSeq[Char] {
+
+  def apply(i: Int): Char = self.charAt(i)
+
+  protected[this] def fromSpecificIterable(coll: strawman.collection.Iterable[Char]): IndexedSeq[Char] =
+    WrappedString.fromSpecific(coll)
+  protected[this] def newSpecificBuilder(): Builder[Char, IndexedSeq[Char]] = WrappedString.newBuilder()
+  def iterableFactory: SeqFactory[IndexedSeq] = ImmutableArray
```

Comment:
Why not use the default `IndexedSeq` implementation?

Reply:
Being a fixed-size array it's the most similar to `String` in spirit